### PR TITLE
Redesign navbar  for mobile first design

### PR DESCRIPTION
**Pull request by brokelyn**
The category page is just an idea to fill the gap. So if you dont like it we can of course take it away.

### CODE REVIEW

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -1,42 +1,13 @@
...
-						>Log out</button
-					>
-				</form>
-			{:else}
-				<NavButton href="/login" text="Log in" />
-				<NavButton href="/register" text="Register" />
-			{/if}
-		</div>
-	</div>
+	<Footer />
```
Comment (nilsalm) [inline src/routes/+layout.svelte:12]: Great improvement!

**Review by nilsalm — CHANGES_REQUESTED** [review]
I think it's well done and a good idea. However i'm not so sure if a footer is the way to go. The main part of the app is adding items to a list and this part is already quite big with the categories and templates and textbox. With having the footer now as well, the space for displaying the items of a list gets quite small which is cutting into the real value of the app. (on smaller screens). 
I also feel that the footer is a bit high maybe and that the emojis dont fit the vibe of the rest of the app. We could leave them for now and make a real design for icons later. 

Could a hamburger/slider menu from the side be an alternative? I'll try to pu some screenshots below

**Comment by nilsalm** [comment]
IMG_3062

**Comment by nilsalm** [comment]
Oh, I got another idea!! Maybe we could just hide the footer while the keyboard (on mobile) is visible? And Hide the keyboard when the user clicks outside of it or submits  and then show the footer again? This would solve the space issue and keep the footer which is so nicely done!

**Comment by nilsalm** [comment]
I think we could even more the "log out" to /profile and get rid of the header-navbar altogether when the user is logged in? What do you think?

**Comment by brokelyn** [comment]
I like having a top bar so maybe we can keep it?

**Comment by brokelyn** [comment]
Regarding the other stuff. We can definitely try to hide the footer when keyboard is open. I dont like having side menu, I think they are quite ugly and only useful if there is more content then three buttons. 
Regarding the item creator/editor. I think the buttons could be a bit small on mobile. I really like that you can access everything directly but once you have multiple template and/or categories it might become less user friendly?
It would like to rework that part as well a bit, but I am not sure how to do it yet :+1:

**Comment by nilsalm** [comment]
Maybe then we keep the top bar instead? 

For the item-adder (we need to find a name!) I agree that the UI needs some more love, maybe smarter ordering could be nice? Also the templates could be less accessible actually because you probably don't user them as often as categories. So one for click to get a modal with all templates would be OK I suppose. So then it would only be the categories left in that area, and if we order them on "last used" could be cool? Or could be really confusing because it's always changing. Currently it's preselecting the category if there is another item (picked or not) in that list with the same-ish name. So if there was Banana in Fruits before then Fruits is being selected automatically as soon as you type "Ban".

**Comment by brokelyn** [comment]
keep the top bar and remove the bottom bar? Would like to have both even though it takes space :D 

Yes templates could get hidden behind a modal or some other button which then brings up the templates :+1: 

for the item editor there also should be two modes (hidden/visible) -> The app has basically two use cases
1. adding things to the list 
2. going shopping with the list

for the first you would need the item editor and for the second it would be cool if its hidden so you have more space for the actual list

**Comment by nilsalm** [comment]
> keep the top bar and remove the bottom bar? Would like to have both even though it takes space :D

Do it, we can see how it feels after a while and still change it later :)

> The app has basically two use cases

I also thought about that but havent come up with a good solution for toggling between the modes yet. I definetely would want to keep it supersimple to toggle that, because sometimes while shopping you get an idea and you quickly want to note it down before forgetting, so that should be easy. 

> Regarding the auto category

My idea with the categories is that you make it your own. You know cheese and milk are in the safe area in the supermarket in germany, but in totally different sections in sweden? The app should not tell you where things are, but you should teach the app your own thought pattern. Only then it can be useful to everybody i believe. Categories can be categories of goods but it can also be sections/aisles in the store that you usually go it. But ofc even for this there could be a setting in the profile if the user wants to use auto-generated standard categories or if they want to do it themselves?



But to wrap this PR up, what do you think about keeping both header and footer navbar now, but look into hiding the footer bar while the keyboard is up?

File: src/lib/types/ItemEditorState.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export enum State {
```
Comment (nilsalm) [inline src/lib/types/ItemEditorState.ts:1]: Maybe this should be called ItemEditorState too?

Comment (nilsalm) [reply src/lib/types/ItemEditorState.ts:1]: Or ItemAdderState (if we settled on that name now?)

**Review by nilsalm — APPROVED** [review]
small name change, maybe

**Pull request merged**